src/net/sourceforge/kolmafia/persistence/ItemDatabase.java

```diff
@@ -52,12 +51,12 @@ public class ItemDatabase {
   private static int maxItemId = 0;
 
   private static String[] canonicalNames = new String[0];
-  private static final IntegerArray useTypeById = new IntegerArray();
-  private static final IntegerArray attributesById = new IntegerArray();
-  private static final IntegerArray priceById = new IntegerArray();
-  private static final IntegerArray nameLength = new IntegerArray();
-  private static final StringArray pluralById = new StringArray();
-  private static final StringArray imageById = new StringArray();
+  private static final Map<Integer, Integer> useTypeById = new LinkedHashMap<>();
+  private static final Map<Integer, Integer> attributesById = new LinkedHashMap<>();
+  private static final Map<Integer, Integer> priceById = new LinkedHashMap<>();
+  private static final Map<Integer, Integer> nameLength = new LinkedHashMap<>();
+  private static final Map<Integer, String> pluralById = new LinkedHashMap<>();
+  private static final Map<Integer, String> imageById = new LinkedHashMap<>();
```

Comment:
Even the names of these - xxxById - make you think they are maps. But, again, are they ever iterated? If not, why not just a HashMap?